libs/x-content/impl/src/main/java/org/elasticsearch/xcontent/provider/json/JsonXContentGenerator.java

```diff
@@ -364,6 +364,31 @@ public void writeString(String value) throws IOException {
         }
     }
 
+    @Override
+    public void writeStringArray(String[] array) throws IOException {
+        try {
+            if (isFiltered()) {
+                // filtered serialization does not work correctly with the bulk array serializer, so we need to fall back to serializing
+                // the array one-by-one
```

Comment:
Do we have a test that catches this case? (to be clear: I am not asking you to write one, just curious if there was one that failed without this)

Reply:
Yup only found this via a test actually, it's a couple of filtering tests that fail here even :)